hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
@mxw This specification is just for Vreg8 instructions right? Others sizes < than 8 bits like a `movl` and `movw` it zero's upper bits? Seems to me that `movl`always zero.

Yes it's a nop. Our problem before, like I show, is that we was incorrectly using it on lowered instructions result on invalid values once the instructions is changed to a nop.

I misinterpreted what you say. But it's clear now. I will update this PR. Thank you for the clarification.